Unify the docker command, command_inject, and command_wrapper

hexylena opened the pull request:
Assigned to Björn to ensure he's happy with the change as well. CC @scholtalbers can you test this with your `sg` setup please?

Instead of three separate options: `command`, `command_inject`, and `command_wrapper`, they've been unified into the `command` option. This makes life easier and it _looks_ more like the docker command you run on the command line, so a little less magic for the admins.

hexylena committed:
Unify the docker command, command_inject, and command_wrapper

hexylena commented:
Realised this would interfere with some of the other changes I plan to make, so putting this in WIP temporarily.

hexylena committed:
Revert command inject and document rationale

jmchilton commented:
The config file should have all the options commented out and the default displayed. The default should be hard coded in the Python and not read from the file. That way if the default behavior changes, these config files should not have to change. You should only have to update the config files you explicitly modify. This is the pattern we try to use with galaxy.ini and I think is pretty common.

hexylena commented:
It is pretty common and you are right :) We've just been bad about this historically. Will update this PR with that.

hexylena committed:
Hardcode defaults, per @jmchilton's comments

jmchilton commented:
Is putting `run` in the command pattern a good idea, or should the framework be responsible for that? I feel like the command wrapper might be useful for other stuff such as pre-fetching images, etc... and not just the run command.

hexylena commented:
@jmchilton, indeed, in an upcoming PR I do a `.replace('run {docker_args}', 'inspect {container_id}')`. I'll remove `run` and just prepend `{docker args}` when necessary I guess. That feels like a marginally better abstraction

hexylena committed:
Move run into python from template

hexylena commented:
How's that @jmchilton

jmchilton commented on lib/galaxy/web/base/interactive_environments.py:
```diff
@@ -65,10 +65,12 @@ def load_deploy_config(self, default_dict={}):
-        default_dict['wx_tempdir'] = 'False'
-        default_dict['command_wrapper'] = ''
+        default_dict = {
+            'command': 'docker run {docker_args}',
```
Still have run here.

hexylena replied:
Fixed.

hexylena committed:
Correct stupid mistake from lack of testing

hexylena committed:
Remove small formatting issue

bgruening committed:
Add docker_galaxy_temp_dir

This is useful for if the normal tmp dir is located on NFS. In this case it's useful to override the temporary directory with an sshfs mount for example.

bgruening committed:
Update ipython.ini.sample

bgruening committed:
Update rstudio.ini.sample

hexylena committed:
Merge pull request #9 from erasche/docker_galaxy_temp_dir

Add docker_galaxy_temp_dir

hexylena commented:
@jmchilton @bgruening has included some code in this PR, so we'll have to have you review it at your leisure :)

bgruening commented:
LGTM, I added my stuff as well. @jmchilton I can not +1 this I guess as my commits are in as well.

jmchilton commented:
(Aside: You can :+1: and merge PRs that contain commits you authored - only the issuers of the PR should (can) not merge it.)

jmchilton commented:
Seems solid to me :+1:.
